sqlmesh/core/macros.py

```diff
@@ -476,6 +478,12 @@ def this_model(self) -> str:
             raise SQLMeshError("Model name is not available in the macro evaluator.")
         return this_model.sql(dialect=self.dialect, identify=True, comments=False)
 
+    @property
+    def this_model_fqn(self) -> str:
```

Comment:
Do we need the same API in the `ExecutionContext`? Also what's wrong with `evaluator.this_model.fqn`?

Reply:
Ah shit, `this_model` is not a model.

Reply:
Shouldn't we expose `@this_model_fqn` then?

Reply:
> Do we need the same API in the ExecutionContext?

I can look into that as well.

> Also what's wrong with evaluator.this_model.fqn?
> Ah shoot, this_model is not a model.
> Shouldn't we expose @this_model_fqn then?

So we actually have `locals["this_model"]` which is a `Table` reference. This is what `evaluator.this_model` is based on. The issue however is that it's [already resolved](https://github.com/TobikoData/sqlmesh/blob/73e2ef4a8d811461b8777eedec96101a710b9790/sqlmesh/core/renderer.py#L135-L140).

Reply:
Right. My question is if we expose this through `evaluator` do we want to expose it directly in SQL too

Reply:
Ah, yeah I was wondering the same, but won't we end up resolving the unresolved version after the macros are rendered? Or am I missing something? In which case, it'd be meaningless to expose that variable in `locals`.

Reply:
I see